src/models/recipe/recipe.schema.ts

```diff
@@ -0,0 +1,35 @@
+import { Schema, model } from "mongoose";
+import mongoose from "mongoose";
+import { IRecipeDocument, IRecipeModel } from "./recipe.types";
+
+//Recipe Schema
+const RecipeSchema: Schema = new Schema(
+  {
+    name: {
+      type: String,
+      required: true,
+    },
+    desription: {
+      type: String,
+      required: true,
+    },
+    likes: {
+      key: String,
+      ref: "user",
+      value: Date,
+    },
```

Comment:
Since we want to store likes as an Object with key value pairs (the keys being the `_id` of all the users liking the recipe) we have to use `Schema.Types.Mixed`
```suggestion
    likes: { type: Schema.Types.Mixed, required: true, default: {} }
```